ui/src/app/shared/services/workflows-service.ts

```diff
@@ -241,21 +241,58 @@ export const WorkflowsService = {
                     }
                     throw new Error('no artifact logs are available');
                 }
-
                 return from(requests.get(this.getArtifactLogsPath(workflow, nodeId, container, archived)));
             }),
             mergeMap(r => r.text.split('\n')),
+            //check for empty content
+            filter(x => !!x),
             map(content => ({content, podName: workflow.status.nodes[nodeId].displayName}) as LogEntry),
             filter(x => !!x.content.match(grep))
         );
     },
 
+    getContainerLogsFromArtifacts(workflow: Workflow, container: string, grep: string, archived: boolean): Observable<LogEntry> {
+        // Iterate workflow and get all node ids and check if node is a pod. Then sort nodes by startedAt
+        const nodeIds: string[] = Object.keys(workflow.status.nodes)
+            .filter(nodeId => {
+                return workflow.status.nodes[nodeId].type === 'Pod';
+            })
+            .sort((a, b) => {
+                return new Date(workflow.status.nodes[a].startedAt).getTime() - new Date(workflow.status.nodes[b].startedAt).getTime();
+            });
+
+        // Iterate over all nodes and get the logs
+        return from(nodeIds).pipe(
+            concatMap(nodeId => {
+                return of(hasArtifactLogs(workflow, nodeId, container)).pipe(
+                    switchMap(isArtifactLogs => {
+                        if (!isArtifactLogs) {
+                            throw new Error('no artifact logs are available');
+                        }
+                        return from(requests.get(this.getArtifactLogsPath(workflow, nodeId, container, archived)));
+                    }),
+                    mergeMap(r => r.text.split('\n')),
+                    //check for empty content
+                    filter(x => !!x),
+                    map(content => ({content, podName: workflow.status.nodes[nodeId].displayName}) as LogEntry),
+                    filter(x => !!x.content.match(grep))
+                );
+            })
+        );
+    },
+
     getContainerLogs(workflow: Workflow, podName: string, nodeId: string, container: string, grep: string, archived: boolean): Observable<LogEntry> {
         const getLogsFromArtifact = () => this.getContainerLogsFromArtifact(workflow, nodeId, container, grep, archived);
+        const getLogsFromArtifacts = () => this.getContainerLogsFromArtifacts(workflow, container, grep, archived);
         const getLogsFromCluster = () => this.getContainerLogsFromCluster(workflow, podName, container, grep);
 
         // If our workflow is archived, don't even bother inspecting the cluster for logs since it's likely
         // that the Workflow and associated pods have been deleted
+        // If we have a node ID, then we can assume that we are looking for logs for a specific node, if not we wan´t the logs from all nodes
```

Comment:
```suggestion
        // If we don't have a node id, then we want the logs from all nodes
```
- fix typo
- shorten quite a bit
- invert the statement -- the first condition is no node id, then existing node id
- be consistent -- "node id" is what you used in other comments in this file